python/lsst/db/mysqlUtils.py

```diff
@@ -0,0 +1,418 @@
+    containing these key/value pairs translated to names accepted by connect()
+    as needed). Hint, to get a subset, do something like:
+    dict = readCredentialFile(fN)
+    (hst, prt, usr, pwd) = [dict[k] for k in ('host', 'port', 'user', 'passwd')].
+    This function only reads from the [mysql] section, e.g., it is not full
+    equivalent to how mysql command like utility which obtains the value from the
+    last occurrence of k in section [mysql] or [client] in the file.
+    """
+    ret = {}
+    if fName.startswith('~'):
+        fName = os.path.expanduser(fName)
+    if not os.path.isfile(fName):
+        raise DbException(DbException.MISSING_OPT_FILE, fName)
+    cnf = ConfigParser.ConfigParser()
+    cnf.read(fName)
+
+    theSection = "mysql"
+    if not cnf.has_section(theSection):
+        raise Exception("Missing section '%s' in '%s'" % (theSection, fName))
+    for o in optionToConnectArgMap:
+        if cnf.has_option(theSection, o):
+            theKey = optionToConnectArgMap.get(o, o)
+            ret[theKey] = cnf.get(theSection, o)
+    return ret
+
+#### Database-related functions ####################################################
+def createDb(conn, dbName, mayExist=False):
+    """
+    Create database <dbName>.
+
+    @param dbName      Database name.
+    @param mayExist    Flag indicating what to do if the database exists.
+
+    Raises DbException if the database already exists and mayExist is False, or if
+    MySQL error occurs.
+
+    Note, it will not connect to that database and it will not make it default.
+    """
+    if dbName is None:
+        raise DbException(DbException.INVALID_DB_NAME, "<None>")
+    try:
+        conn.execute("CREATE DATABASE `%s`" % _sanitize(dbName))
+    except DBAPIError as e:
+        if e.orig.args[0] == MySqlErr.ER_DB_CREATE_EXISTS:
+            if not mayExist:
+                raise DbException(DbException.DB_EXISTS, dbName)
+        else:
+            raise DbException(DbException.DRIVER_ERROR,
+                              str(e.orig.args[0]), e.orig.args[1])
+
+def useDb(conn, dbName):
+    """
+    Connect to database <dbName>.
+
+    @param dbName     Database name.
+
+    Raises DbException if MySQL error occurs.
+    """
+    try:
+        conn.execute("USE `%s`" % _sanitize(dbName))
+    except DBAPIError as e:
+        if e.orig.args[0] == MySqlErr.ER_BAD_DB_ERROR:
+            raise DbException(DbException.DB_DOES_NOT_EXIST, dbName)
+        else:
+            raise DbException(DbException.DRIVER_ERROR,
+                              str(e.orig.args[0]), e.orig.args[1])
+
+def dbExists(conn, dbName):
+    """
+    Return True if database <dbName> exists, False otherwise.
+
+    @param dbName     Database name.
+
+    Raises DbException if MySQL error occurs.
+    """
+    cmd = "SELECT COUNT(*) FROM information_schema.schemata "
+    cmd += "WHERE schema_name = :dbN"
+    try:
+        return conn.execute(text(cmd), {"dbN": dbName}).first()[0] == 1
```

Comment:
for count queries especially, you can use:
`conn.execute(text(cmd), {"dbN": dbName}).scalar() == 1`

This takes the first column of the first row.